apps/condo/domains/common/utils/url.utils.js

```diff
@@ -42,10 +42,11 @@ function extractRootDomain (url) {
 
 // Removes the path from the link, leaving the part with subdomains and protocol
 function extractOrigin (url) {
-    const doubleSlashIndex = url.indexOf('//')
-    const startIndex = doubleSlashIndex > 1 ? doubleSlashIndex + 2 : 0
-    const firstSeparator = url.indexOf('/', startIndex)
-    return url.substring(0, firstSeparator > -1 ? firstSeparator : undefined)
+    try {
+        return new URL(url).origin
+    } catch {
+        return null
```

Comment:
don't we want to log errors here ?

Reply:
I guess there's no need...since it's only used it on front-end we control urls via admin panel